### DISCUSSION TITLE
Old Spark Streaming version support

### DISCUSSION THREAD

**Discussion by AliGouta** [stackabletech/spark-k8s-operator#287] [Q&A]
Hello,

We currently have spark streaming applications compiled and released with version 2.3.0 and a target kubernetes cluster version set to 1.24. Do you think that the spark operator supports old versions ?

Thanks !

**Comment by sbernauer** [comment]
Hello @AliGouta,

thanks for your interest!
We are aware that migrating your Spark jobs to a new Spark version is a humongous task.
However, we do not explicitly support or plan on supporting older versions but it might work.

In case you are interested in giving it a try yourself, here is the [relevant Dockerfile](https://github.com/stackabletech/docker-images/blob/main/spark-k8s/Dockerfile) and [definition of the versions to build](https://github.com/stackabletech/docker-images/blob/2e9d38507d83460606e0035cd05f4c8164515f3a/conf.py#L253). You might be able to swap out the Spark version.

For the Kubernetes version, besides it not being not officially supported, I think the current Stackable versions (23.4 and lower) should work, as I am not aware of the use of any fancy features that were added after K8s 1.24.

All the best,
Sebastian
